### PR TITLE
support `readOnly` backend in LC client/data dbs

### PR DESCRIPTION
**Pull request by etan-status**
When accessing DB in `readOnly` mode that does not already have latest schema, initial writes trigger `attempt to write a readonly database`. Avoid that by only writing schema when DB is not `readOnly`, and provide data from legacy tables if such are present.

### CODE REVIEW

File: beacon_chain/light_client_db.nim [hunk 1]
```diff
@@ -221,6 +261,8 @@ func putSyncCommittee*(
     db: LightClientDB, period: SyncCommitteePeriod,
     syncCommittee: altair.SyncCommittee) =
   doAssert period.isSupportedBySQLite
+  if distinctBase(db.syncCommittees.putStmt) == nil:
```
Comment (arnetheduck) [inline beacon_chain/light_client_db.nim]: this should probably be an assert, at least that's what we've done elsewhere - ie the higher-level code might be assuming writing happened and get into a messy state

**Review by arnetheduck — APPROVED** [review]

File: beacon_chain/light_client_db.nim (hunk 1 above)
Comment (etan-status) [reply beacon_chain/light_client_db.nim]: higher-level code can't assume that, as there may be concurrent writes from different process, or other DB corruption.

But agree, having an assert to highlight API misuse doesn't hurt.

Comment (etan-status) [reply beacon_chain/light_client_db.nim]: would go even further and assert on `not readOnly` - the current condition would hold in readonly mode (where tables already exist)

Comment (arnetheduck) [reply beacon_chain/light_client_db.nim]: > higher-level code can't assume that, as there may be concurrent writes from different process, or other DB corruption.

not really, de facto we require exclusive writing access to the sqlite db while the bn is running - the readonly mode is an expression of that: the code (in chaindag in particular), as written, doesn't work with multiple writers.

Comment (arnetheduck) [reply beacon_chain/light_client_db.nim]: ..and we assert on every other form of corruption we detect

Comment (etan-status) [reply beacon_chain/light_client_db.nim]: Got it!

**Comment by arnetheduck** [comment]
btw, we do have tests for the readonly mode - clearly, they don't cover the light client stuff though, probably a good idea to add some so we don't regress (since the readonly mode is used in utilities mainly, not the main app)

**Pull request merged**